### ISSUE TITLE
'Set as Default Style' is overridden by adhoc style changes

### ISSUE THREAD

**Issue by dsample** [jgraph/drawio#2164]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
After setting a default style for connectors, adhoc changes to any connector's style will also change the default style.

To my knowledge this affects both lines and shapes, and all style properties.

**To Reproduce**
1. Create a diagram with two shapes
2. Connect the shapes
3. Change the connector's style (eg. set colour to blue)
4. Click `Set as Default Style`
5. Add another connector to the shapes. _The default style (blue) is used._
6. Change the style of one of the connectors (eg. change colour to red)
7. Add another connector to the shapes.
8. See bug - The default style was not used (the new line is red when it should be blue)

**Expected behavior**
When setting a default style for a line/shape I expect it to preserve that default style until I click `Set as Default Style` again to update the default.

**Screenshots**
<img width="257" alt="image" src="https://user-images.githubusercontent.com/34761/127978715-d2e2e2fa-5e47-4af7-944c-abc9aa528c36.png">

**draw.io version (In the Help->About menu of the draw.io editor):**

- 14.9.2

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Firefox
 - Version: 91.0b8

**Comment by mararad** [comment]
Hi,

I created the video reproducing your steps and all is ok at my place. I tested with drawio 14.9.4, Chrome Version 92.0.4515.131 (Official Build) (64-bit) and Firefox  90.0.2 (64-bit) 
Win 10 Pro

If we performed the same steps, my suggestion would be to update drawio to the latest and check again. 
Close all tabs, load https://app.diagrams.net/clear.html and then load diagrams.net again

https://user-images.githubusercontent.com/44292306/128333737-1017b19a-5a1b-44ef-88c0-304e78c3fc85.mp4

Regards,

**Comment by dsample** [comment]
> If we performed the same steps

Sorry, but you haven't performed the same steps. Using the style presets alone does not change the default, but if you change one of the properties of an existing item, such as the line colour, that will incorrectly change the default.

https://user-images.githubusercontent.com/34761/128355646-4b5fed8e-efd9-439c-9685-6702bf4bd71f.mov

**Comment by mararad** [comment]
Thanks for the video. I reproduced this.

**Comment by dsample** [comment]
This bug is still present in the current version (I encounter it almost every day)

**Comment by dsample** [comment]
> This is not a bug, it's the intended behaviour.

Could you explain how it's intended behaviour?

The 'set as default style' button's behaviour of overridden by a simple selection of colour or line style. I can't see how that's intended.

**Comment by awojtas** [comment]
I think I see this also when I'm pulling in a shape from the library. If I've changed the font of an existing shape in my diagram, the font and formatting of the shape will be broken.

Here's an example shape, before changing any formatting:
![image](https://user-images.githubusercontent.com/3030232/173214701-4670b8ae-91d7-47e1-ae01-3efe100b29c0.png)

And now I've changed the font of a text box in the diagram (where the red arrow is pointing to), and the shape in the library has changed.

![image](https://user-images.githubusercontent.com/3030232/173214770-f5b64ee6-131c-4719-bebe-230242f9dfec.png)

This gets worse - if I change fill and other formatting, the shapes get to the point of being uselessly messed up.
Fortunately, closing draw.io and reopening it fixes it. Until next time.

This super annoying!

Draw.io 19.0.0 on Windows 10
